File: mobile_apps/android_app/TigerWallet/app/src/main/java/com/tigerwallet/app/ui/screens/StakingScreen.kt
package com.tigerwallet.app.ui.screens

import android.os.Bundle
import android.widget.Toast
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp

// Staking Screen - Stake and earn
class StakingScreen : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            MaterialTheme {
                StakingScreenContent()
            }
        }
    }
}

@OptIn(ExperimentalMaterialApi::class)
@Composable
fun StakingScreenContent() {
    var selectedTab by remember { mutableStateOf("Stake") }
    var stakeAmount by remember { mutableStateOf("") }
    var selectedPool by remember { mutableStateOf("ETH 2.0") }
    var isStaking by remember { mutableStateOf(false) }
    
    val tabs = listOf("Stake", "Earn", "Pools")
    val pools = listOf(
        StakingPool("ETH 2.0", "4.2%", "1.5 ETH", "0.063 ETH"),
        StakingPool("BNB", "3.8%", "0 BNB", "0 BNB"),
        StakingPool("SOL", "6.5%", "0 SOL", "0 SOL"),
        StakingPool("MATIC", "5.2%", "0 MATIC", "0 MATIC")
    )
    
    val context = LocalContext.current

    Scaffold(
        topBar = {
            TopAppBar(title = { Text("Staking") })
        }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
        ) {
            // Tab Selector
            TabRow(selectedTabIndex = tabs.indexOf(selectedTab)) {
                tabs.forEachIndexed { index, tab ->
                    Tab(
                        selected = selectedTab == tab,
                        onClick = { selectedTab = tab },
                        text = { Text(tab) }
                    )
                }
            }

            when (selectedTab) {
                "Stake" -> stakeView(stakeAmount, { stakeAmount = it }, selectedPool, { selectedPool = it }, isStaking, { isStaking = it }, pools, context)
                "Earn" -> earnView(pools)
                "Pools" -> poolsView(pools)
            }
        }
    }
}

@Composable
fun stakeView(
    amount: String,
    onAmountChange: (String) -> Unit,
    pool: String,
    onPoolChange: (String) -> Unit,
    isLoading: Boolean,
    onLoadingChange: (Boolean) -> Unit,
    pools: List<StakingPool>,
    context: android.content.Context
) {
    LazyColumn(
        modifier = Modifier
            .fillMaxSize()
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        // Pool Selector
        item {
            OutlinedCard {
                Column(modifier = Modifier.padding(16.dp)) {
                    Text("Select Pool", style = MaterialTheme.typography.subtitle2)
                    Spacer(modifier = Modifier.height(8.dp))
                    pools.forEach { p ->
                        Row(
                            modifier = Modifier.fillMaxWidth(),
                            horizontalArrangement = Arrangement.SpaceBetween
                        ) {
                            Text(p.name)
                            Text("APY: ${p.apy}", color = Color.Green)
                        }
                    }
                }
            }
        }

        // Amount
        item {
            OutlinedCard {
                Column(modifier = Modifier.padding(16.dp)) {
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween
                    ) {
                        Text("Amount", style = MaterialTheme.typography.subtitle2)
                        TextButton(onClick = { onAmountChange("1.0") }) {
                            Text("MAX")
                        }
                    }
                    OutlinedTextField(
                        value = amount,
                        onValueChange = onAmountChange,
                        modifier = Modifier.fillMaxWidth(),
                        placeholder = { Text("0.0") },
                        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Decimal),
                        singleLine = true
                    )
                }
            }
        }

        // Stake Button
        item {
            Button(
                onClick = {
                    if (amount.isEmpty()) {
                        Toast.makeText(context, "Enter amount", Toast.LENGTH_SHORT).show()
                        return@Button
                    }
                    onLoadingChange(true)
                    android.os.Handler(android.os.Looper.getMainLooper()).postDelayed({
                        onLoadingChange(false)
                        Toast.makeText(context, "Staked successfully!", Toast.LENGTH_SHORT).show()
                    }, 2000)
                },
                modifier = Modifier
                    .fillMaxWidth()
                    .height(56.dp),
                enabled = !isLoading
            ) {
                if (isLoading) {
                    CircularProgressIndicator(modifier = Modifier.size(24.dp), color = Color.White)
                } else {
                    Text("Stake", style = MaterialTheme.typography.title6)
                }
            }
        }
    }
}

@Composable
fun earnView(pools: List<StakingPool>) {
    LazyColumn(
        modifier = Modifier.fillMaxSize().padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        items(pools) { pool ->
            Card(modifier = Modifier.fillMaxWidth()) {
                Column(modifier = Modifier.padding(16.dp)) {
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween
                    ) {
                        Column {
                            Text(pool.name, style = MaterialTheme.typography.subtitle1)
                            Text("APY: ${pool.apy}", color = Color.Green)
                        }
                        Column(horizontalAlignment = Alignment.End) {
                            Text("Staked", style = MaterialTheme.typography.caption)
                            Text(pool.staked)
                        }
                    }
                    Spacer(modifier = Modifier.height(12.dp))
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Column {
                            Text("Pending Reward", style = MaterialTheme.typography.caption)
                            Text(pool.reward, color = Color.Green)
                        }
                        Button(onClick = { }) {
                            Text("Claim")
                        }
                    }
                }
            }
        }
    }
}

@Composable
fun poolsView(pools: List<StakingPool>) {
    LazyColumn(
        modifier = Modifier.fillMaxSize().padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        items(pools) { pool ->
            Card(modifier = Modifier.fillMaxWidth()) {
                Column(modifier = Modifier.padding(16.dp)) {
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Text(pool.name, style = MaterialTheme.typography.h6)
                        Column(horizontalAlignment = Alignment.End) {
                            Text(pool.apy, style = MaterialTheme.typography.h5, color = Color.Green, fontWeight = androidx.compose.ui.text.font.FontWeight.Bold)
                            Text("APY", style = MaterialTheme.typography.caption)
                        }
                    }
                    Spacer(modifier = Modifier.height(8.dp))
                    Text("Total Staked: ${pool.staked}", style = MaterialTheme.typography.caption)
                    Spacer(modifier = Modifier.height(12.dp))
                    Button(onClick = { }, modifier = Modifier.fillMaxWidth()) {
                        Text("Stake")
                    }
                }
            }
        }
    }
}

data class StakingPool(
    val name: String,
    val apy: String,
    val staked: String,
    val reward: String
)

File: mobile_apps/ios_app/TigerWallet/Screens/StakingScreen.swift
import SwiftUI

// Staking Screen - Stake and earn
struct StakingScreen: View {
    @State private var selectedTab: String = "Stake"
    @State private var stakeAmount: String = ""
    @State private var selectedPool: String = "ETH 2.0"
    @State private var isStaking: Bool = false
    
    let tabs = ["Stake", "Earn", "Pools"]
    
    let stakingPools = [
        StakingPool(name: "ETH 2.0", apy: "4.2%", staked: "1.5 ETH", reward: "0.063 ETH"),
        StakingPool(name: "BNB", apy: "3.8%", staked: "0 BNB", reward: "0 BNB"),
        StakingPool(name: "SOL", apy: "6.5%", staked: "0 SOL", reward: "0 SOL"),
        StakingPool(name: "MATIC", apy: "5.2%", staked: "0 MATIC", reward: "0 MATIC")
    ]
    
    var body: some View {
        NavigationView {
            VStack(spacing: 0) {
                // Tab Selector
                HStack(spacing: 0) {
                    ForEach(tabs, id: \.self) { tab in
                        Button(action: { selectedTab = tab }) {
                            VStack {
                                Text(tab)
                                    .fontWeight(selectedTab == tab ? .bold : .regular)
                                Rectangle()
                                    .fill(selectedTab == tab ? Color.orange : Color.clear)
                                    .frame(height: 2)
                            }
                        }
                        .frame(maxWidth: .infinity)
                    }
                }
                .padding()
                
                ScrollView {
                    VStack(spacing: 20) {
                        if selectedTab == "Stake" {
                            stakeView
                        } else if selectedTab == "Earn" {
                            earnView
                        } else {
                            poolsView
                        }
                    }
                    .padding()
                }
            }
            .navigationTitle("Staking")
        }
    }
    
    var stakeView: some View {
        VStack(spacing: 20) {
            // Pool Selector
            VStack(alignment: .leading, spacing: 8) {
                Text("Select Pool")
                    .font(.subheadline)
                    .foregroundColor(.secondary)
                
                Menu {
                    ForEach(stakingPools, id: \.name) { pool in
                        Button(action: { selectedPool = pool.name }) {
                            Text("\(pool.name) - \(pool.apy)")
                        }
                    }
                } label: {
                    HStack {
                        Text(selectedPool)
                            .fontWeight(.semibold)
                        Spacer()
                        if let pool = stakingPools.first(where: { $0.name == selectedPool }) {
                            Text("APY: \(pool.apy)")
                                .font(.caption)
                                .foregroundColor(.green)
                        }
                        Image(systemName: "chevron.down")
                    }
                    .padding()
                    .background(Color.gray.opacity(0.1))
                    .cornerRadius(8)
                }
            }
            
            // Amount
            VStack(alignment: .leading, spacing: 8) {
                Text("Amount")
                    .font(.subheadline)
                    .foregroundColor(.secondary)
                
                HStack {
                    TextField("0.0", text: $stakeAmount)
                        .keyboardType(.decimalPad)
                        .font(.title)
                    
                    Button(action: { stakeAmount = "1.0" }) {
                        Text("MAX")
                            .foregroundColor(.orange)
                    }
                }
                .padding()
                .background(Color.gray.opacity(0.1))
                .cornerRadius(8)
            }
            
            // Stake Button
            Button(action: stake) {
                HStack {
                    if isStaking {
                        ProgressView()
                            .progressViewStyle(CircularProgressViewStyle(tint: .white))
                    } else {
                        Text("Stake")
                            .fontWeight(.bold)
                    }
                }
                .frame(maxWidth: .infinity)
                .padding()
                .background(Color.orange)
                .foregroundColor(.white)
                .cornerRadius(12)
            }
            .disabled(isStaking)
        }
    }
    
    var earnView: some View {
        VStack(spacing: 16) {
            ForEach(stakingPools, id: \.name) { pool in
                earnCard(pool: pool)
            }
        }
    }
    
    func earnCard(pool: StakingPool) -> some View {
        VStack(spacing: 12) {
            HStack {
                VStack(alignment: .leading) {
                    Text(pool.name)
                        .font(.headline)
                    Text("APY: \(pool.apy)")
                        .font(.caption)
                        .foregroundColor(.green)
                }
                Spacer()
                VStack(alignment: .trailing) {
                    Text("Staked")
                        .font(.caption)
                        .foregroundColor(.secondary)
                    Text(pool.staked)
                        .font(.subheadline)
                }
            }
            
            HStack {
                VStack(alignment: .leading) {
                    Text("Pending Reward")
                        .font(.caption)
                        .foregroundColor(.secondary)
                    Text(pool.reward)
                        .font(.subheadline)
                        .foregroundColor(.green)
                }
                Spacer()
                Button(action: {}) {
                    Text("Claim")
                        .font(.caption)
                        .padding(.horizontal, 16)
                        .padding(.vertical, 8)
                        .background(Color.orange)
                        .foregroundColor(.white)
                        .cornerRadius(8)
                }
            }
        }
        .padding()
        .background(Color.gray.opacity(0.1))
        .cornerRadius(12)
    }
    
    var poolsView: some View {
        VStack(spacing: 16) {
            ForEach(stakingPools, id: \.name) { pool in
                poolCard(pool: pool)
            }
        }
    }
    
    func poolCard(pool: StakingPool) -> some View {
        VStack(spacing: 12) {
            HStack {
                Text(pool.name)
                    .font(.headline)
                Spacer()
                Text(pool.apy)
                    .font(.title2)
                    .fontWeight(.bold)
                    .foregroundColor(.green)
                Text("APY")
                    .font(.caption)
                    .foregroundColor(.secondary)
            }
            
            HStack {
                Text("Total Staked: \(pool.staked)")
                    .font(.caption)
                    .foregroundColor(.secondary)
                Spacer()
                Button(action: {}) {
                    Text("Stake")
                        .font(.caption)
                        .padding(.horizontal, 16)
                        .padding(.vertical, 8)
                        .background(Color.orange)
                        .foregroundColor(.white)
                        .cornerRadius(8)
                }
            }
        }
        .padding()
        .background(Color.gray.opacity(0.1))
        .cornerRadius(12)
    }
    
    func stake() {
        guard !stakeAmount.isEmpty else { return }
        isStaking = true
        DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
            isStaking = false
            stakeAmount = ""
        }
    }
}

struct StakingPool: Identifiable {
    let id = UUID()
    let name: String
    let apy: String
    let staked: String
    let reward: String
}

#Preview {
    StakingScreen()
}
